neo/rawio/neuronexusrawio.py

```diff
@@ -0,0 +1,275 @@
+"""
+Neuronexus has their own file format based on their Allego Recording System
+https://www.neuronexus.com/webinar/allego-software-updates/
+
+The format involves 3 files:
+  * The *.xdat.json metadata file
+  * The *_data.xdat binary file of all raw data
+  * The *_timestamps.xdat binary file of the timeestamp data
+  
+Based on sample data is appears that the binary file is always a float32 format
+Other information can be found within the metadata json file
+
+
+The metadata file has a pretty complicated structure as far as I can tell
+a lot of which is dedicated to probe information, which won't be handle at the 
+the Neo level.
+
+It appears that the metadata['status'] provides most of the information necessary
+for generating the initial memory map (global sampling frequency), n_channels,
+n_samples.
+
+metadata['sapiens_base']['biointerface_map'] provides all the channel specific information
+like channel_names, channel_ids, channel_types.
+
+An additional note on channels. It appears that analog channels are called `pri` or
+`ai0` within the metadata where as digital channels are called `din0` or `dout0`.
+In this first implementation it is up to the user to do the appropriate channel slice
```

Comment:
What do you mean by this?

Reply:
This was a note for myself. I don't know what the naming convention is for neuronexus (like what does pri actually stand for?). So what do we want to expose to the end user. Their `pri` names or their `ai0` names. We could do either?

Reply:
Ah, got you, I thought you wer talking about slicing across channels.

Reply:
Sorry yeah the plexon issue with unclear channel naming was on my mind. So since we aren't do sub-streaming yet both the analog and digital channels all come out in the same stream (or buffer since I like the buffer term better). So the end-user needs to understand the channel naming scheme if they want to slice out the digital data and only look at the amplifier data.

Reply:
Ah, perfect, that clarifies it all. So in the new terminology you are providing buffer based streams and once someone with more expertise on the format comes along then we can improve it to use "logic" based streams. Right?

Reply:
Yeah basically. I could technically figure it out, but since this is a buffer I figured I would leave as is for now to see what Sam was thinking of for the stream api to minimize the surface area of fixes. If a user complains then I could just implement something myself.